ledger-core/virtual/integration/deactivate_test.go

```diff
@@ -394,10 +391,241 @@ func TestVirtual_CallMethod_On_DeactivatedDirtyState(t *testing.T) {
+						require.Equal(t, objectRef, ctx.Request.Callee)
+						require.Equal(t, []byte(origDirtyMem), ctx.ObjectDescriptor.Memory())
+					},
+					&execution.Update{
+						Type:     execution.OutgoingCall,
+						Result:   requestResult,
+						Outgoing: execution.NewRPCBuilder(outgoing, objectRef).Deactivate(),
+					},
+				).AddContinue(
+					func(result []byte) {
+						synchronizeExecution.Synchronize()
+					},
+					&execution.Update{
+						Type:   execution.Done,
+						Result: requestresult.New([]byte("finish Deactivate"), objectRef),
+					},
+				)
+			}
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+			typedChecker.VStateRequest.Set(func(req *payload.VStateRequest) bool {
+				require.Equal(t, p1, req.AsOf)
+				require.Equal(t, objectRef, req.Object)
+
+				flags := payload.RequestLatestDirtyState | payload.RequestLatestValidatedState |
+					payload.RequestOrderedQueue | payload.RequestUnorderedQueue
+				require.Equal(t, flags, req.RequestedContent)
+
+				content := &payload.VStateReport_ProvidedContentBody{
+					LatestDirtyState: &payload.ObjectState{
+						Reference: reference.Local{},
+						Class:     class,
+						State:     []byte(origDirtyMem),
+					},
+					LatestValidatedState: &payload.ObjectState{
+						Reference: reference.Local{},
+						Class:     class,
+						State:     []byte(origValidatedMem),
+					},
+				}
+
+				report := payload.VStateReport{
+					Status:          payload.Ready,
+					AsOf:            req.AsOf,
+					Object:          objectRef,
+					ProvidedContent: content,
+				}
+				server.SendPayload(ctx, &report)
+				return false // no resend msg
+			})
+
+			typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+				require.Equal(t, objectRef, res.Callee)
+				require.Equal(t, outgoing, res.CallOutgoing)
+				require.Equal(t, []byte("finish Deactivate"), res.ReturnArguments)
+				return false // no resend msg
+			})
+			// we must resend all messages below for creating right SMObject in next pulse
+			typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+				require.Equal(t, objectRef, report.Object)
+				require.Equal(t, payload.Ready, report.Status)
+				require.True(t, report.DelegationSpec.IsZero())
+				require.Equal(t, int32(0), report.UnorderedPendingCount)
+				require.Equal(t, int32(1), report.OrderedPendingCount)
+				require.NotNil(t, report.ProvidedContent)
+				require.NotNil(t, report.ProvidedContent.LatestDirtyState)
+				require.False(t, report.ProvidedContent.LatestDirtyState.Deactivated)
+				require.Equal(t, []byte(origDirtyMem), report.ProvidedContent.LatestDirtyState.State)
+				require.NotNil(t, report.ProvidedContent.LatestValidatedState)
+				require.False(t, report.ProvidedContent.LatestValidatedState.Deactivated)
+				require.Equal(t, []byte(origDirtyMem), report.ProvidedContent.LatestValidatedState.State)
+				return true
+			})
+			typedChecker.VDelegatedCallRequest.SetResend(true)
```

Comment:
Is it necessary to resend DelegatedCall? Can we not test SMDelegatedTokenRequest here?

Reply:
Yes, it is better to use automatic processing, because it is difficult to create all the internal tables manually. This is why I resend lots of messages to system.